### ISSUE TITLE
PickupMethod::getId throws when updating cart gateway

### ISSUE THREAD

**Issue by Tam** [mediabeastnz/craft-commerce-click-and-collect#2]
Getting the ID from a PickupMethod throws an exception when updating a carts gateway (posting `gatewayId` to `commerce/cart/update-cart`).

The line in question:
https://github.com/mediabeastnz/craft-commerce-click-and-collect/blob/master/src/models/PickupMethod.php#L68

```
craft\commerce\errors\NotImplementedException in /var/www/html/vendor/mediabeastnz/craft-commerce-click-and-collect/src/models/PickupMethod.php:68
Stack trace:
#0 /var/www/html/vendor/craftcms/commerce/src/elements/Order.php(1678): mediabeastnz\ClickAndCollect\models\PickupMethod->getId()
#1 /var/www/html/vendor/craftcms/commerce/src/services/Orders.php(197): craft\commerce\elements\Order->getShippingMethodId()
#2 /var/www/html/vendor/craftcms/commerce/src/controllers/BaseFrontEndController.php(60): craft\commerce\services\Orders->cartArray(Object(craft\commerce\elements\Order))
#3 /var/www/html/vendor/craftcms/commerce/src/controllers/CartController.php(382): craft\commerce\controllers\BaseFrontEndController->cartArray(Object(craft\commerce\elements\Order))
#4 /var/www/html/vendor/craftcms/commerce/src/controllers/CartController.php(340): craft\commerce\controllers\CartController->_returnCart()
#5 [internal function]: craft\commerce\controllers\CartController->actionUpdateCart()
#6 /var/www/html/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#7 /var/www/html/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#8 /var/www/html/vendor/craftcms/cms/src/web/Controller.php(187): yii\base\Controller->runAction('update-cart', Array)
#9 /var/www/html/vendor/yiisoft/yii2/base/Module.php(528): craft\web\Controller->runAction('update-cart', Array)
#10 /var/www/html/vendor/craftcms/cms/src/web/Application.php(299): yii\base\Module->runAction('commerce/cart/u...', Array)
#11 /var/www/html/vendor/craftcms/cms/src/web/Application.php(565): craft\web\Application->runAction('commerce/cart/u...', Array)
#12 /var/www/html/vendor/craftcms/cms/src/web/Application.php(278): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#13 /var/www/html/vendor/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest(Object(craft\web\Request))
#14 /var/www/html/web/index.php(22): yii\base\Application->run()
#15 {main}
```

**Comment by green17** [comment]
Is this a Craft Commerce 3 problem?

**Comment by john-henry** [comment]
@Tam This should fix this for you https://github.com/mediabeastnz/craft-commerce-click-and-collect/pull/4

**Comment by mylesthedev** [comment]
Closed with #4
